### PR TITLE
feature/stock_reservations

### PR DESCRIPTION
**Pull request by mkcosta**
RELEASE NOTE:
* added `stock_reservations` docs

### CODE REVIEW

File: stock_reservations/README.md [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+Stock reservations are used to reserve the stock associated with an order's line items of type `sku` or `bundle` until the order is approved.
+
+{% hint style="warning" %}
+Active stock reservations (if any) determine the actual availability of the associated [stock items](../stock\_items/).
+{% endhint %}
+
+### Manual stock reservation before order placement
+
+You can temporarily reserve the stock associated with a line item before the order placement by sending a specific [trigger attribute](../line\_items#reserving-the-stock) when adding the line item to the order. This way the necessary stock reservations are created in `draft` status, with an expiration date of **1 hour** (you can change the default value at the [inventory model](../inventory\_models#stock-reservation-settings) level). If the order isn't placed before the expiration date, the corresponding reserved stock is released keeping the stock reservations in draft. If the line item is removed, the corresponding reserved stock is released and the status of any associated active stock reservation is moved to `cancelled`.
```
Comment (mkcosta) [inline stock_reservations/README.md:17]: `If the line item is removed, the corresponding reserved stock is released and the status of any associated active stock reservation is moved to `cancelled`.`
Not true, if a `line_item` is deleted the associated `stock_reservations` get destroyed and the reserved stock is released. The status is moved to `cancelled` only when the `order` gets `cancelled`.

File: stock_reservations/README.md [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+{% hint style="info" %}
+Stock reservations act on the whole line item quantity. If the available stock is split between different stock locations, multiple stock reservations are created.
+{% endhint %}
+
+### Automatic stock reservation at order placement
+
+When the order is placed any existing active stock reservation associated with its line items is moved from `draft` to `pending`. If no draft stock reservation is associated with the order's line items, the stock reservations needed to block the whole order's stock are automatically created in `pending` status and the associated stock is reserved without decrementing the stock item quantities. Once the order is approved the stock reservations status is moved to `completed` and the stock item quantities are decremented. If the order is cancelled before approval the stock reservations status is moved to `cancelled` and the reserved stock is released becoming available again.
```
Comment (mkcosta) [inline stock_reservations/README.md:25]: We do not mention that `stock_reservations` expire only when on `draft` status. Dunno if makes sense, maybe just to assure that when the `order` is placed the stock is reserved until it moves to `approved`.

Comment (sfiorucci) [reply stock_reservations/README.md:25]: Fixed.

File: stock_reservations/README.md (hunk 1 above)
Comment (sfiorucci) [reply stock_reservations/README.md]: Fixed.

File: stock_reservations/README.md [hunk 3]
```diff
@@ -14,15 +14,15 @@ Active stock reservations (if any) determine the actual availability of the asso
...
+You can temporarily reserve the stock associated with a line item before the order placement by sending a specific [trigger attribute](../line\_items#reserving-the-stock) when adding the line item to the order. This way the necessary stock reservations are created in `draft` status, with an expiration date of **1 hour** (you can change the default value at the [inventory model](../inventory\_models#stock-reservation-settings) level). If the order isn't placed before the expiration date, the corresponding reserved stock is released keeping the stock reservations in draft. If the line item is removed, the corresponding reserved stock is released and any associated active stock reservation is deleted.
```
Comment (mkcosta) [inline stock_reservations/README.md:17]: Allora, le expired `stock_reservations` vengono in realtà brasate dal job, quindi eviterei di dire che rimangano in `draft`: vengono distrutte e nello stesso istante viene rilasciato lo stock. La granularità della suddetta azione dipende da un job che gira ogni minuto e/o appena dopo la place di qualsiasi ordine. Questo dovrebbe ridurre al minimo inconsistenze di stock (quest'ultima parte la ometterei, diciamo solo che lo per le expired si rilascia lo stock e vengono cancellate).

File: stock_reservations/README.md [hunk 4]
```diff
@@ -14,15 +14,15 @@ Active stock reservations (if any) determine the actual availability of the asso
...
+When the order is placed any existing active stock reservation associated with its line items is moved from `draft` to `pending`. If no draft stock reservation is associated with the order's line items, the stock reservations needed to block the whole order's stock are automatically created in `pending` status and the associated stock is reserved without decrementing the stock item quantities. Once the order is approved, the stock item quantities are decremented. If the order is cancelled before approval, the reserved stock is released becoming available again. In both cases, the involved stock reservations are deleted.
```
Comment (mkcosta) [inline stock_reservations/README.md:25]: Aggiungerei: `Once the order is approved, the stock item quantities are decremented and the reserved stock released.`

**Pull request merged**